### PR TITLE
Rework win_setup_template and update Python tests

### PR DESCRIPTION
**Pull request by MaksimZhukov**

### CODE REVIEW

File: builders/NixPythonBuilder.psm1 [hunk 1]
```diff
@@ -18,7 +18,7 @@ class NixPythonBuilder : PythonBuilder {
 
         $this.InstallationTemplateName = "nix_setup_template.sh"
         $this.InstallationScriptName = "setup.sh"
-        $this.PythonConfigScriptLocation = "../tests/sources/python_config.py"
+        $this.PythonConfigScriptLocation = "../tests/python_config.py"
```
Comment (vmapetr) [inline builders/NixPythonBuilder.psm1]: $this.PythonConfigScriptLocation = Join-Path -Path $PSScriptRoot -ChildPath "../tests/python_config.py"

File: azure-pipelines/test_job.yml [hunk 2]
```diff
@@ -40,13 +41,13 @@ jobs:
             }
           }
           Invoke-Pester -Script $pesterParams -OutputFile "test_results.xml" -OutputFormat NUnitXml
-        workingDirectory: '$(Build.SourcesDirectory)/tests/sources'
+        workingDirectory: '$(Build.SourcesDirectory)/tests'
```
Comment (maxim-lobanov) [inline azure-pipelines/test_job.yml]: indent is incorrect

File: tests/Python.Tests.ps1 [hunk 3]
```diff
@@ -19,34 +19,30 @@ Describe "Python toolcache tests" {
...
-    It "Run sample code" {
-        Get-CommandExitCode -Command "python ./main.py" | Should -Be 0
+    It "Run simple code" {
+        Get-CommandExitCode -Command "python ./simple_test.py" | Should -Be 0
     }
 
-    It "Validate modules"  {
-        if (IsNixPlatform $Platform) {
+    if (IsNixPlatform $Platform) {
+        It "Check if all the python modules are installed"  {
```
Comment (maxim-lobanov) [inline tests/Python.Tests.ps1]: `Check if all the python modules are installed`

File: tests/Python.Tests.ps1 [hunk 4]
```diff
@@ -19,34 +19,30 @@ Describe "Python toolcache tests" {
...
             Get-CommandExitCode -Command "python ./python_modules.py" | Should -Be 0
         }
-    }
 
-    It "Check Tkinter" {
-        if (IsNixPlatform $Platform) {
+        It "Check Tkinter" {
```
Comment (maxim-lobanov) [inline tests/Python.Tests.ps1]: `Check Tkinter module is available`

File: tests/Python.Tests.ps1 [hunk 5]
```diff
@@ -19,34 +19,30 @@ Describe "Python toolcache tests" {
...
         }
-    }
 
-    It "Validate Pyinstaller" {
-        # Pyinstaller 3.5 does not support Python 3.8.0. Check issue https://github.com/pyinstaller/pyinstaller/issues/4311
-        if ($Version -lt "3.8.0") {
-            Get-CommandExitCode "pip install pyinstaller" | Should -Be 0
-            Get-CommandExitCode -Command "pyinstaller --onefile ./main.py" | Should -Be 0
-            Get-CommandExitCode "./dist/main" | Should -Be 0
+        It "Check if shared libraries was linked correctly" {
```
Comment (maxim-lobanov) [inline tests/Python.Tests.ps1]: `Check if shared libraries are linked correctly`

File: tests/psutil_install_test.sh [hunk 6]
```diff
@@ -1,3 +1,4 @@
+# Make sure shared libraries was linked correctly
```
Comment (maxim-lobanov) [inline tests/psutil_install_test.sh]: `Check if shared libraries are linked correctly`

File: installers/win_setup_template.ps1 [hunk 7]
```diff
@@ -1,8 +1,13 @@
 $Architecture = "{{__ARCHITECTURE__}}"
-$Version = "{{__VERSION__}}"
+[Version]$Version = "{{__VERSION__}}"
```
Comment (vmapetr) [inline installers/win_setup_template.ps1]: Let's add [String] to $Architecture and $PythonExecName to be consistent

File: installers/win_setup_template.ps1 [hunk 8]
```diff
@@ -13,36 +18,66 @@ function Get-ArchitectureFilter {
...
     param(
         [Parameter (Mandatory = $true)]
-        [String]$ArchFilter
+        [String]$ArchFilter,
+        [Parameter (Mandatory = $true)]
+        [String]$Architecture,
+        [Parameter (Mandatory = $true)]
+        [Boolean]$IsMSI,
+        [Parameter (Mandatory = $true)]
+        [Int32]$MajorVersion,
```
Comment (vmapetr) [inline installers/win_setup_template.ps1]: Why not simple [Version]$version?

File: builders/NixPythonBuilder.psm1 (hunk 1 above)
Comment (MaksimZhukov) [reply builders/NixPythonBuilder.psm1]: Thanks!

File: azure-pipelines/test_job.yml (hunk 2 above)
Comment (MaksimZhukov) [reply azure-pipelines/test_job.yml]: Thanks!

File: tests/Python.Tests.ps1 (hunk 3 above)
Comment (MaksimZhukov) [reply tests/Python.Tests.ps1]: Thanks!

File: tests/Python.Tests.ps1 (hunk 4 above)
Comment (MaksimZhukov) [reply tests/Python.Tests.ps1]: Thanks!

File: tests/Python.Tests.ps1 (hunk 5 above)
Comment (MaksimZhukov) [reply tests/Python.Tests.ps1]: Thanks!

File: tests/psutil_install_test.sh (hunk 6 above)
Comment (MaksimZhukov) [reply tests/psutil_install_test.sh]: Thanks!

File: installers/win_setup_template.ps1 (hunk 7 above)
Comment (MaksimZhukov) [reply installers/win_setup_template.ps1]: Thanks!

**Pull request merged**